eos/db/saveddata/queries.py

```diff
@@ -521,6 +521,8 @@ def save(stuff):
 def remove(stuff):
     removeCachedEntry(type(stuff), stuff.ID)
     with sd_lock:
+        # Merge the object to make sure we don't already have it in session
+        saveddata_session.merge(stuff)
```

Comment:
I'm curious about this... do you remember what was going on that caused an issue here?

> Merge our session to make sure we aren't trying to work with a different instance of the same object.

I can't think of anywhere where we are working on a different instance of the same object... Once an Item is loaded, it hangs around in memory. Once a Fit is loaded, it hangs around in memory (unless deleted, which I guess this might solve?).